cranelift/codegen/src/isa/x64/lower.rs

```diff
@@ -3138,8 +3138,11 @@ fn lower_insn_to_regs<C: LowerCtx<I = Inst>>(
                     ctx.emit(Inst::gen_move(dst, src, output_ty));
 
                     // Generate an all 1s constant in an XMM register. This uses CMPPS but could
-                    // have used CMPPD with the same effect.
+                    // have used CMPPD with the same effect. Note, we zero the temp we allocate
+                    // because if not, there is a chance that the register we use could be initialized
+                    // with NaN .. in which case the CMPPS would fail since NaN != NaN.
                     let tmp = ctx.alloc_tmp(output_ty).only_reg().unwrap();
+                    ctx.emit(Inst::xmm_rm_r(SseOpcode::Xorps, RegMem::from(tmp), tmp));
                     let cond = FcmpImm::from(FloatCC::Equal);
```

Comment:
Looks like this should have been `FloatCC::UnorderedOrGreaterThanOrEqual` which uses `FcmpImm::UnorderedOrGreaterThanOrEqual` to encode to the `0x05` immediate. Using that might avoid the extra `XORPS`.

Reply:
In the old backend we were using `PCMPEQB` to generate the "all ones vector" so this wasn't a problem, but I think in this new backend we should check around and make sure no more of the "all ones vectors" we create will have this same issue. Before Christmas I was working on a PR to simplify constant emission so that we only have one place to check... looks like I should have finished that.

Reply:
Yes, my understanding (after some more reading) is that the i8x16 (pcmpeqb) variant is more canonical; would prefer that over reasoning about FP comparison edge cases. +1 to centralizing this logic!

Reply:
> UnorderedOrGreaterThanOrEqual

Yes, I suppose any unordered instruction that also checks for equality should work. Just depends on what that actually lowers to and the supported CPU feature. That encoding to 0x05 ( NLT_US) looks good except it signals. Not really sure what that means for cranelift .. maybe @cfallin @abrown you have comment. Perhaps instead we'd want to encode as 0x15 or NLT_UQ which does not signal.

Reply:
https://www.felixcloutier.com/x86/cmppd#tbl-3-1

Reply:
That does look better 👍